WordPress/src/main/java/org/wordpress/android/util/WPHtml.java

```diff
@@ -449,22 +450,20 @@ public static String getContent(WPImageSpan imageSpan) {
             String localBlogID = imageSpan.getMediaFile().getBlogId();
             Blog currentBlog = WordPress.wpDB.instantiateBlogByLocalId(Integer.parseInt(localBlogID));
             // If it's not a gif and blog don't keep original size, there is a chance we need to resize
-            if (currentBlog != null && !mediaFile.getMimeType().equals("image/gif") &&
-                !currentBlog.getMaxImageWidth().equals("Original Size")) {
-                int maxImageWidth = Integer.parseInt(currentBlog.getMaxImageWidth());
-                // use the correct resize settings.
-                width = Math.min(width, maxImageWidth);
+            if (currentBlog != null && !mediaFile.getMimeType().equals("image/gif")
+                    && MediaUtils.getImageWidthSettingFromString(currentBlog.getMaxImageWidth()) != Integer.MAX_VALUE) {
+                width = MediaUtils.getMaximumImageWidth(width, currentBlog.getMaxImageWidth());
```

Comment:
Just a note here, the older code was keeping the `0` for `width` while the newer code will have `width` become `1024`.